modules/gitops-creating-user-defined-cluster-roles-and-configuring-user-defined-permissions-for-application-controller.adoc

```diff
@@ -0,0 +1,201 @@
+
+As a cluster administrator, to add user-defined permissions to your aggregated cluster role, you must create one or more user-defined cluster roles and configure user-defined permissions for Argo CD Application Controller of a cluster-scoped Argo CD instance.
+
+.Prerequisites
+
+* You have enabled the creation of aggregated cluster roles for the Argo CD Application Controller component of a cluster-scoped Argo CD instance.
+* You have the following default cluster roles that are created and managed by the {gitops-title} Operator:
++
+** `<argocd_name>-<argocd_namespace>-argocd-application-controller` aggregated cluster role
+** `<argocd_name>-<argocd_namespace>-argocd-application-controller-view`
+** `<argocd_name>-<argocd_namespace>-argocd-application-controller-admin` 
+
+.Procedure
+
+. Create a new cluster role with the required labels and permissions by using the following command:
++
+[source,terminal]
+----
+$ oc apply -n <namespace> -f <cluster_role_name>.yaml
+----
++
+where:
+
+`<namespace>`:: Specifies the name of your defined namespace.
+`<cluster_role_name>`:: Specifies the name of your defined cluster role YAML file.
++
+.Example user-defined cluster role YAML
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata: 
+  name: user-application-controller # <1>
+  labels: # <2>
+    app.kubernetes.io/managed-by: spring-petclinic
+    app.kubernetes.io/name: example
+    app.kubernetes.io/part-of: argocd
+    argocd/aggregate-to-admin: 'true'
+rules: # <3>
+  - verbs:
+      - '*'
+    apiGroups:
+      - ''
+    resources:
+      - namespaces
+      - persistentvolumeclaims
+      - persistentvolumes
+      - configmaps
+  - verbs:
+      - '*'
+    apiGroups:
+      - compliance.openshift.io
+    resources:
+      - scansettingbindings
+----
+<1> The name of the user-defined cluster role.
+<2> The labels match the predefined list of an existing `<argocd_name>-<argocd_namespace>-argocd-application-controller-admin` cluster role.
+<3> The user-defined permissions that are to be added into the aggregated cluster role through the `<argocd_name>-<argocd_namespace>-argocd-application-controller-admin` cluster role.
++
+[TIP]
+====
+Alternatively, you can use the web console to create a user-defined cluster role from the *Administrator* perspective. You can go to *User Management* -> *Roles* -> *Create Role*, use the preceding YAML template to add permissions, and click *Create*.
+====
++
+.Example output
+[source,terminal]
+----
+clusterrole.rbac.authorization.k8s.io/user-application-controller created
+----
++
+A user-defined cluster role is created. 
+
+. Verify that the `<argocd_name>-<argocd_namespace>-argocd-application-controller-admin` cluster role inherits permissions from the user-defined cluster role, by running the following command:
++
+[source,terminal]
+----
+$ oc get ClusterRole/<argocd_name>-<argocd_namespace>-argocd-application-controller-admin -o yaml
```

Comment:
need a "where" sequence for user-replaceable values

Reply:
Addressed